browser/browsing_data/brave_clear_browsing_data_browsertest.cc

```diff
@@ -0,0 +1,383 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/path_service.h"
+#include "base/run_loop.h"
+#include "base/single_thread_task_runner.h"
+#include "base/task/post_task.h"
+#include "base/test/scoped_feature_list.h"
+#include "brave/browser/browsing_data/brave_clear_browsing_data.h"
+#include "chrome/app/chrome_command_ids.h"
+#include "chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/browser/profiles/profile_window.h"
+#include "chrome/browser/search_engines/template_url_service_factory.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_commands.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/browser_list.h"
+#include "chrome/browser/ui/browser_list_observer.h"
+#include "chrome/common/chrome_paths.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/search_test_utils.h"
+#include "chrome/test/base/testing_browser_process.h"
+#include "chrome/test/base/testing_profile.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/browsing_data/core/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/browsing_data_remover.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/common/content_features.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/test_navigation_observer.h"
+
+using content::BraveClearBrowsingData;
+using content::WebContents;
+
+namespace {
+
+class BrowserRemovedObserver : public BrowserListObserver {
+ public:
+  BrowserRemovedObserver() : removed_browser_(nullptr) {
+    BrowserList::AddObserver(this);
+    original_browsers_count_ = BrowserList::GetInstance()->size();
+  }
+
+  ~BrowserRemovedObserver() override {
+    BrowserList::RemoveObserver(this);
+  }
+
+  Browser* CheckReturn() const {
+    EXPECT_EQ(original_browsers_count_ - 1, chrome::GetTotalBrowserCount());
+    return removed_browser_;
+  }
+
+  // Wait for a new browser to be removed.
+  Browser* WaitForBrowserRemoval() {
+    if (removed_browser_)
+      return CheckReturn();
+    run_loop_.Run();
+    EXPECT_NE(removed_browser_, nullptr);
+    return CheckReturn();
+  }
+
+ private:
+  // BrowserListObserver
+  void OnBrowserRemoved(Browser* browser) override {
+    removed_browser_ = browser;
+    run_loop_.Quit();
+  }
+
+  size_t original_browsers_count_;
+  Browser* removed_browser_;
+  base::RunLoop run_loop_;
+
+  DISALLOW_COPY_AND_ASSIGN(BrowserRemovedObserver);
+};
+
+}  // namespace
+
+class BraveClearDataOnExitTest
+    : public InProcessBrowserTest,
+      public BraveClearBrowsingData::OnExitTestingCallback {
+ public:
+  void SetUpOnMainThread() override {
+    BraveClearBrowsingData::SetOnExitTestingCallback(this);
```

Comment:
Maybe also `BraveClearBrowsingData::SetOnExitTestingCallback(nullptr);` in teardown?

Reply:
Done.